master/internal/task/rendezvous.go

```diff
@@ -165,16 +155,21 @@ func (r Rendezvous) rank(id cproto.ID) int {
 // trial, once all the containers are ready this function will return true afterward because this
 // function is used in deciding if the trial should be forcibly killed when terminating.
 func (r *Rendezvous) ready() bool {
+	if r == nil {
```

Comment:
true, it was just for safety, but true

Reply:
Okay. I am actually wrong. This can happen if r is Rendezvous-typed nil, this can totally run.

Reply:
ah yeah, it's not possible in that, there is no code path that can call it now, but i have it there for nil safety since it's called in a reentrant state machine